lib/react.dart

```diff
@@ -204,37 +251,102 @@ abstract class Component {
+  ///
+  /// See: <https://facebook.github.io/react/docs/react-component.html#updating-shouldcomponentupdate>
+  bool shouldComponentUpdateWithContext(Map nextProps, Map nextState, Map nextContext) => null;
+
   /// ReactJS lifecycle method that is invoked immediately before rendering when [nextProps] or [nextState] are being
   /// received.
   ///
   /// This method is not called for the initial [render].
   ///
   /// Use this as an opportunity to perform preparation before an update occurs.
   ///
+  /// __Note__: Choose either this method or [componentWillUpdateWithContext]. They are both called at the same time so
+  /// using both provides no added benefit.
+  ///
   /// See: <https://facebook.github.io/react/docs/react-component.html#updating-componentwillupdate>
   void componentWillUpdate(Map nextProps, Map nextState) {}
 
+  /// ReactJS lifecycle method that is invoked immediately before rendering when [nextProps], [nextState], or
+  /// [nextContext] are being received.
+  ///
+  /// This method is not called for the initial [render].
+  ///
+  /// Use this as an opportunity to perform preparation before an update occurs.
+  ///
+  /// __Note__: Choose either this method or [componentWillUpdate]. They are both called at the same time so using both
+  /// provides no added benefit.
+  ///
+  /// See: <https://facebook.github.io/react/docs/react-component.html#updating-componentwillupdate>
+  void componentWillUpdateWithContext(Map nextProps, Map nextState, Map nextContext) {}
+
   /// ReactJS lifecycle method that is invoked immediately after the `Component`'s updates are flushed to the DOM.
   ///
   /// This method is not called for the initial [render].
   ///
   /// Use this as an opportunity to operate on the [rootNode] (DOM) when the `Component` has been updated as a result
   /// of the values of [prevProps] / [prevState].
   ///
+  /// __Note__: Choose either this method or [componentDidUpdateWithContext]. They are both called at the same time so
+  /// using both provides no added benefit.
+  ///
   /// See: <https://facebook.github.io/react/docs/react-component.html#updating-componentdidupdate>
   void componentDidUpdate(Map prevProps, Map prevState) {}
 
+  /// ReactJS lifecycle method that is invoked immediately after the `Component`'s updates are flushed to the DOM.
+  ///
+  /// This method is not called for the initial [render].
+  ///
+  /// Use this as an opportunity to operate on the [rootNode] (DOM) when the `Component` has been updated as a result
+  /// of the values of [prevProps] / [prevState] /[prevContext].
+  ///
+  /// __Note__: Choose either this method or [componentDidUpdate]. They are both called at the same time so using both
+  /// provides no added benefit.
+  ///
+  /// See: <https://facebook.github.io/react/docs/react-component.html#updating-componentdidupdate>
+  void componentDidUpdateWithContext(Map prevProps, Map prevState, Map prevContext) {}
```

Comment:
Since React 16 doesn't include `prevContext` in `componentDidUpdate`, can we just omit this lifecycle method?